### PR TITLE
Orange Pi 5 / RK3588 Support

### PR DESCRIPTION
**Pull request by brooksytech**
Initial support for the Orange Pi 5 (RK3588 SBC)

### CODE REVIEW

File: scripts/mkimage [hunk 1]
```diff
@@ -94,6 +94,13 @@ if [ "${DISK_LABEL}" = "gpt" ]; then
 fi
 sync
 
+# set partition 3 as boot for u-boot.
+if [ "${PARTITION_TABLE}" = "gpt" ]; then
```
[the remark this answers is not in the document]
Comment (brooksytech) [reply scripts/mkimage:98]: This isnt the u-boot partition. This is the partition exlinux.conf lives on.

File: projects/Rockchip/bootloader/mkimage [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2017-2020 Team LibreELEC (https://libreelec.tv)
+# Copyright (C) 2021-present Fewtarius
+
+source ${PROJECT_DIR}/${PROJECT}/devices/${DEVICE}/options
+
+case "${PKG_SOC}" in
```
[the remark this answers is not in the document]
Comment (brooksytech) [reply projects/Rockchip/bootloader/mkimage]: Resolved

File: projects/Rockchip/bootloader/update.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    dtb=$(basename $all_dtb)
+      echo -n "Updating $dtb... "
+      cp -p $SYSTEM_ROOT/usr/share/bootloader/$dtb $BOOT_ROOT &>/dev/null
+      echo "done"
+  done
+
+# update bootloader
+
+MYDEV=$(awk '/^Hardware/ {print $4}' /proc/cpuinfo)
+case ${MYDEV} in
```
[the remark this answers is not in the document]
Comment (brooksytech) [reply projects/Rockchip/bootloader/update.sh:86]: Resolved

File: projects/Rockchip/devices/RK3588/device.config [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+DEVICE_PLAYBACK_PATH_HP="HP"
+UI_SERVICE="weston.service"
+DEVICE_BRIGHTNESS="128"
+SPLASH_LOADER="plymouth"
+
+# GPIOS
+DEVICE_TEMP_SENSOR="/sys/devices/virtual/thermal/thermal_zone*/temp"
+
+# FREQ Governors
+CPU_FREQ=("/sys/devices/system/cpu/cpufreq/policy0" "/sys/devices/system/cpu/cpufreq/policy4")
```
[the remark this answers is not in the document]
Comment (brooksytech) [reply projects/Rockchip/devices/RK3588/device.config]: Resolved

**Comment by brooksytech** [comment]
I think things are in a good state now:
*Games correctly launching from ES
*HDMI audio correctly set at boot
*All usb ports working
*Wayland / Weston / Desktop mode working
*Ethernet works
*Updating works
*Revamped Rockchip bootloader
*Fix u-boot install and config for RK3588 boards

Missing / broken features:
*Add additional wifi and bluetooth drivers. (None of my adapters seem to work)
*Console messages don't work (init & update messages)
*Dolphin broken in Panfork
*Slow shutdown / reboot time (Appears to be a kernel issue, other OS report issue as well)
*Bring back some cores / emus purged with anbernic support. 
*Update to Mesa and mainline kernel if/when available.
*Add SPL boot support (for nvme ssd booting) and ability to flash from jelos (console only)
*Break out ES_systems for cores not supported on arm devices. 

Untested:
*USB C video out (hardware should support this).
*NVME / Sata as game storage drives.

**Comment by fewtarius** [comment]
Nice work!  We should be able to re-use a lot of this for my 3588 device as well. :)

**Pull request merged**